websockets/handlers/msg_channel_wsh.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python
+import json
+import logging
+import urllib
+import threading
+import traceback
+from queue import Empty
+
+from mod_pywebsocket import stream, msgutil
+from wptserve import stash as stashmod
+
+logger = logging.getLogger()
+
+address, authkey = stashmod.load_env_config()
+stash = stashmod.Stash("msg_channel", address=address, authkey=authkey)
+
+def log(uuid, msg):
+    logger.debug(f"{uuid}: {msg}")
+
+
+def web_socket_do_extra_handshake(request):
+    return
+
+
+def web_socket_transfer_data(request):
+    uuid, direction = parse_request(request)
+    log(uuid, f"Got web_socket_transfer_data {direction}")
+
+    with stash.lock:
+        value = stash.take(uuid)
+        if value is None:
+            queue = stash.get_queue()
+            if direction == "read":
+                has_reader = True
+                writer_count = 0
+            else:
+                has_reader = False
+                writer_count = 1
+        else:
+            queue, has_reader, writer_count = value
+            if direction == "read":
+                if has_reader:
+                    raise ValueError("Tried to start multiple readers for the same queue")
```

Comment:
Shouldn't `has_reader` be set to True after this? Or what sets it to True in the stash?

Reply:
Good catch.